### PR TITLE
v2.6.x: Rev libcalico-go and pin go-build to v0.8

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

- rev libcalico-go to v1.7.2
- rev other components
- update go-build to v0.8 (golang 1.8.5)
- back out some tests that were added to master before we cut v2.6 even though they were for post v2.6 function (named ports)
- tweak Makefile to be able to build/test v2.6

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: _data/versions.yml [hunk 1]
```diff
@@ -1,4 +1,52 @@
 v2.6:
+- title: v2.6.3
+  note: |
+    22 November 2017
```
Comment (bcreane) [inline _data/versions.yml]: probably 27 november now

File: _data/versions.yml [hunk 2]
```diff
@@ -1355,17 +1403,17 @@ master:
   note: ""
   components:
      felix:
-      version: 2.x-series
+      version: 2.6.0-8-gecbd512d
```
Comment (bcreane) [inline _data/versions.yml]: whoa, this is scary, good catch!

File: _data/versions.yml [hunk 3]
```diff
@@ -1355,17 +1403,17 @@ master:
...
       url: ""
      calicoctl:
       version: v1.x-series
       url: "https://www.projectcalico.org/builds/calicoctl"
       download_url: "https://www.projectcalico.org/builds/calicoctl"
      calico/node:
-      version: v2.x-series
+      version: v2.6.x-series
```
Comment (bcreane) [inline _data/versions.yml]: shouldn't this be a specific hash?

File: _data/versions.yml [hunk 4]
```diff
@@ -1355,17 +1403,17 @@ master:
...
       url: ""
      calico/cni:
-      version: v1.11.0
+      version: v1.11.1
```
Comment (bcreane) [inline _data/versions.yml]: shouldn't this stay at 1.11.0? https://github.com/projectcalico/cni-plugin/releases/tag/v1.11.0

File: _data/versions.yml [hunk 5]
```diff
@@ -1355,17 +1403,17 @@ master:
...
       url: ""
       download_calico_url: https://github.com/projectcalico/cni-plugin/releases/download/v1.11.0/calico
```
Comment (bcreane) [inline _data/versions.yml]: if the bump to 1.11.1 was intentional, these urls would need to change too.

File: calico_node/Makefile [hunk 6]
```diff
@@ -1,60 +1,31 @@
...
 
+# Read current stream version from _data/versions.yml
+RELEASE_STREAM?=$(shell cat $(VERSIONS_FILE) | $(YAML_CMD) read - "currentReleaseStream")
+
 ###############################################################################
 # Determine whether there's a local yaml installed or use dockerized version.
 # Note in order to install local (faster) yaml: "go get github.com/mikefarah/yaml"
 YAML_CMD:=$(shell which yaml || echo docker run --rm -i $(CALICO_BUILD) yaml)
 
-# Read current stream version from _data/versions.yml
```
Comment (bcreane) [inline calico_node/Makefile]: I'm not sure these changes should be part of the release ... @neiljerram and I have been discussing some changes to the version reporting, but should probably be a separate commit (unless Neil is approving this).

File: calico_node/Makefile [hunk 7]
```diff
@@ -63,6 +34,17 @@ SYSTEMTEST_CONTAINER?=calico/test:$(SYSTEMTEST_CONTAINER_VER)
 # Ensure that the dist directory is always created
 MAKE_SURE_BIN_EXIST := $(shell mkdir -p dist)
 
+.PHONY: print-version
+print-version:
+	$(info $$RELEASE_STREAM is ${RELEASE_STREAM})
+    $(info $$BIRD_VER is ${BIRD_VER})
```
Comment (bcreane) [inline calico_node/Makefile]: these info statements probably need a <tab> indent, not sure how gmake interprets actions with space indents.

File: calico_node/Makefile [hunk 8]
```diff
@@ -250,11 +221,11 @@ dist/allocate-ipip-addr: $(ALLOCATE_IPIP_FILES) vendor
 	mkdir -p dist
 	mkdir -p .go-pkg-cache
 	docker run --rm \
-	-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
-	-v $(CURDIR)/.go-pkg-cache:/go/pkg/:rw \
-	-v $(CURDIR):/go/src/$(PACKAGE_NAME):ro \
-	-v $(VERSIONS_FILE):/versions.yaml:ro \
-	-e VERSIONS_FILE=/versions.yaml \
+  	-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
```
Comment (bcreane) [inline calico_node/Makefile]: indents look wonky - initial space then tab?

File: calico_node/Makefile [hunk 9]
```diff
@@ -527,8 +498,8 @@ semaphore:
 	# Actually run the tests (refreshing the images as required), we only run a
 	# small subset of the tests for testing SSL support.  These tests are run
 	# using "latest" tagged images.
-	$(MAKE) RELEASE_STREAM=master $(NODE_CONTAINER_NAME) st
-	ST_TO_RUN=tests/st/policy $(MAKE) RELEASE_STREAM=master st-ssl
+	$(MAKE) RELEASE_STREAM=$(RELEASE_STREAM) $(NODE_CONTAINER_NAME) st
```
Comment (bcreane) [inline calico_node/Makefile]: this fix is long over due!

File: calico_node/Makefile [hunk 10]
```diff
@@ -558,7 +529,7 @@ release: clean
 	# Check that the images container the right sub-components
 	docker run $(NODE_CONTAINER_NAME) calico-felix --version
 	docker run $(NODE_CONTAINER_NAME) libnetwork-plugin -v
-
+
```
Comment (bcreane) [inline calico_node/Makefile]: extra tabs here and line 541

File: _data/versions.yml (hunk 3 above)
Comment (fasaxc) [reply _data/versions.yml]: This is the "master" section.  I think it's supposed to float at the "master" for this development branch.

File: calico_node/Makefile (hunk 6 above)
Comment (fasaxc) [reply calico_node/Makefile]: This was an accidental revert that happened when I reverted another PR.  Fixed up.

File: calico_node/Makefile (hunk 9 above)
Comment (caseydavenport) [reply calico_node/Makefile]: This fix is, IMO, fine for this branch, but not correct for master.
